src/test/java/uk/gov/pay/api/utils/mocks/ConnectorMockClient.java

```diff
@@ -168,13 +168,10 @@ public void respondOk_whenCreateCharge(int amount, String gatewayAccountId, Stri
                                 settlementSummary,
                                 cardDetails,
                                 validGetLink(chargeLocation(gatewayAccountId, chargeId), "self"),
-                                validGetLink(nextUrl(chargeTokenId), "next_url"), validPostLink(nextUrlPost(), "next_url_post", "application/x-www-form-urlencoded",
-                                        new HashMap<String, String>() {{
-                                            put("chargeTokenId", chargeTokenId);
-                                        }}))));
+                                validGetLink(nextUrl(chargeTokenId), "next_url"),
+                                validPostLink(nextUrlPost(), "next_url_post", "application/x-www-form-urlencoded", getChargeIdTokenMap(chargeTokenId)))));
```

Comment:
This change is to not instantiate the map using the instance initializer, as this cannot be deserialized by JSON.
We've left a comment on the `getChargeIdTokenMap` method about this.

More info can be found here:
https://stackoverflow.com/questions/45876600/hashmap-to-gson-with-maps-as-values